app/controllers/api/status_controller.rb

```diff
@@ -35,14 +35,6 @@ def check
     if params[:score]
       fields.push :score
     end
-    render json: @projects.to_json({
-      only: fields,
-      methods: [:package_manager_url, :stars, :forks, :keywords, :latest_download_url],
-      include: {
-        versions: {
-          only: [:number, :published_at]
-        }
-      }
-    })
+    render json: @projects, each_serializer: ProjectStatusSerializer, show_score: params[:score], show_stats: internal_api_key?
```

Comment:
Should there be a `.joins(:repository_maintenance_stats)` to avoid N+1 queries?

Reply:
I experimented a bit with the different ways to query these. A `join` will cause a lot of extra queries as we are referencing the stat attributes later on and doesn't seem recommended. Adding the stats to `includes` seem to have no real effect, it generates a separate select query for `repository`, `versions`, and the `maintenance_stats`. If use `references` on these relationships, it retrieves all data in one query using `left outer` joins. I'm not sure the one query would result in much difference in performance. I've added the `maintenance_stats` to the includes portion of the queries, but I don't think it had much of an effect on the generated queries.